### ISSUE TITLE
Bug: Users can receive a mention notification when they're no longer a member of the channel/board

### ISSUE THREAD

**Issue by ogi-m** [mattermost/focalboard#1879]
## Steps to reproduce the behavior

1. Have a channel with User-1 and User-2
2. As User-1, open the associated board
3. Open a card and click on the comment field
4. Type in "@" and make sure User-1 is listed
5. As User-2, leave the channel that's associated with the board
6. As User-1, type in "@" in the comment field again (without refreshing)
7. Select User-1 from the autocomplete and post the comment - User-1 receives a notification for the mention

## Expected behavior

Users shouldn't be able to receive notifications for boards they no longer belong to

## Edition and Platform

 - Edition: Mattermost plugin
 - Version: v0.11
 - Browser and OS: Chrome & Firefox

## Additional context

User-1 can only see the card title, the card doesn't unfurl and they can't access it but I would expect them to not get a notification at all.
Also, if the board is refreshed then the User-1 no longer shows up in autocomplete.

**Comment by sbishel** [comment]
@asaadmahmood to think about design for this.

**Comment by asaadmahmood** [comment]
3 things.
1. This is a bug, as it first shows a user that is no longer in the channel associated with the board.
2. And then it also sends a notification.
3. In the future, when we allow mentioning users across the system, we should show a header above the users who are not part of the board. And that should probably be fine for now.

![Boards Card Detail](https://user-images.githubusercontent.com/11034289/144441882-15e5f291-9795-418d-89d7-8fd0fc2acb50.png)
![Boards Card Detail-1](https://user-images.githubusercontent.com/11034289/144441889-437b2bce-3e0c-42a4-935f-c956537c4478.png)

**Comment by asaadmahmood** [comment]
That UI is only application when we implement the role based permissions and detach channels from boards. Right now, the user directly should be removed from the autocomplete list.

**Comment by ogi-m** [comment]
Do we want to split the autocomplete part into a separate issue? That only happens if the board isn't refreshed after the user is removed from the channel and before trying to at-mention. If the board is refreshed, only members will be shown in the autocomplete.
However, they can still manually type out the username of a user outside of the channel and the notification will be send. Those are two separate issues I believe.

**Comment by sbishel** [comment]
@wiggin77 - assigning to you, I think there are both server and front end fixes needed. Although the server should be fixed here. https://github.com/mattermost/focalboard/pull/1216

**Comment by wiggin77** [comment]
@asaadmahmood, @sbishel, just to confirm the desired behaviour... when someone types an @mention for a user that is not a member of the channel we will simply not send the notification.  No error will be displayed on front-end?

**Comment by sbishel** [comment]
Sorry @wiggin77 just saw the question when notified of the PR. I think at a minimum, the user should not appear in the autocomplete and the server should not send...

As a P2, it would be nice to display an error to the user in a similar manner channel's does. But unfortunately we don't really have a mechanism for that.  

Actually, now seeing that the autocomplete list is only if the user has been removed, I'd be willing to wait on a fix for that part, since it will be going from channels -> teams next release. But make sure it's handled correctly for teams.

**Comment by wiggin77** [comment]
> I think at a minimum, the user should not appear in the autocomplete and the server should not send...

Ok, thanks.  I think the PR satisfies the current requirements.
